addons/account/views/account_move_views.xml

```diff
@@ -608,6 +608,13 @@
                          attrs="{'invisible': ['|', ('display_inactive_currency_warning', '=', False), ('move_type', 'not in', ('out_invoice', 'out_refund', 'out_receipt'))]}">
                         In order to validate this invoice, you must <button class="oe_link" type="object" name="action_activate_currency" style="padding: 0; vertical-align: baseline;">activate the currency of the invoice</button>. The journal entries need to be computed by Odoo before being posted in your company's currency.
                     </div>
+                    <!-- Bank account mismatch when importing invoice -->
+                    <div class="alert alert-warning" role="alert" style="margin-bottom:0px;"
+                         attrs="{'invisible':['|', '|', ('state', '!=', 'draft'), ('imported_bank_id', '=', False), ('imported_bank_active', '=', True)]}">
```

Comment:
I'm not sure I understand the use case you point out. When a bank account is created, it is assigned to the partner but set as inactive until the user explicitly confirm that the account is correct. When this happens, the new account is set as active and assigned to this move, but BOTH accounts are kept on the partner. When an invoice is imported later with either account, it will be accepted, which is sensible because companies often have multiple bank accounts.

We do NOT replace the bank account, we only allow the new one if the user confirms it.